public/js/viewer.js

```diff
@@ -64,7 +64,28 @@ var View = (function () {
     }
   }
 
-  var nodes = {
+
+  /**
+   * nodes is an array of THREE.Vector3's.
+   * Returns THREE.Vector3 with position at the midpoint of all nodes
+   */
+  const midPointVector = (nodes) => {
```

Comment:
@dacerron is there a preferred style we should use? I.e. ES5 vs ES6?

Reply:
We use ES5 functions at the component level because when interpreted they are exposed by A-Frame as functions available to other components through the DOM and keeping ``this`` to the DOM element instead of the A-Frame component declaration is important (try to change ``init`` or ``update`` functions in A-Frame components to ES6 and they will break because ``this`` is different), but for local private functions and callback handlers ES6 is nicer and makes more sense. I am also open to recommendations if you think there is some part of the codebase that could be improved by leveraging either syntax. 

Also, for ``const`` vs ``var``, for the modules that we built so far we have just been using var since we are using vanilla modules so var declarations are kept within the function scope, but I have been thinking that as the project grows we might want to switch over to es6 modules, in which case it will be better for us to use const to keep the global scope clean. Again, you are the first person on this team besides me with familiarity with javascript syntax so I welcome your feedback. If in your past work you have come across a comprehensive syntax guide (like the PSRs for PHP https://www.php-fig.org/psr/psr-12/) Then I would be happy to adopt and enforce it.

Reply:
sweet thanks for the explanation! will play around with it some more to get a better understanding.

In terms of style guides, I've been following this one: https://github.com/airbnb/javascript
It's quite long but pretty heavily adopted. Mostly follow the conventions using a linter, maybe we can set something up for this project?